src/BenchmarkDotNet/ListBenchmarks/AsciiTreeDiagram.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+
+namespace BenchmarkDotNet.ListBenchmarks {
+
+    internal class AsciiTreeDiagram
+    {
+        // Constants for drawing lines and spaces
+        private const string Cross = " ├─";
+        private const string Corner = " └─";
+        private const string Vertical = " │ ";
+        private const string Space = "   ";
+            
+        public void PrintNode(Node node, string indent)
+        {
+            Console.WriteLine((string) node.Name);
+
+            // Loop through the children recursively, passing in the
+            // indent, and the isLast parameter
+            var numberOfChildren = node.Children.Count;
+            for (var i = 0; i < numberOfChildren; i++)
+            {
+                var child = node.Children[i];
+                var isLast = (i == (numberOfChildren - 1));
+                PrintChildNode(child, indent, isLast);
+            }
+        }
+
+        void PrintChildNode(Node node, string indent, bool isLast)
+        {
+            // Print the provided pipes/spaces indent
```

Comment:
nit: this comment is not necessary. When I see `Console.Write(ident)` I know that we are printing `indent` because the method argument has a descriptive name